### PR TITLE
BarrageSubscription/Snapshot Throw TableDataException Instead of Incoming gRPC StatusRuntimeException

### LINKED ISSUE #5417: BarrageTables on the server must not propagate upstream errors to downstream clients
When a BarrageTable has a gRPC error, it currently passes that error to downstream clients. This can result in UNAVAILABLE errors (due to network issues upstream) being passed to downstream clients, leading them to believe that they should reconnect.

### PR DESCRIPTION
**Pull request by nbauernfeind**
Fixes #5417

### CODE REVIEW

File: java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java [hunk 1]
```diff
@@ -161,7 +162,7 @@ public void onError(final Throwable t) {
                     .append(t).endl();
 
             // this error will always be propagated to our CheckForCompletion#onError callback
-            resultTable.handleBarrageError(t);
```
Comment (devinrsmith) [inline java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java:164]: "Barrage snapshot error"?

File: java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java [hunk 2]
```diff
@@ -161,7 +162,7 @@ public void onError(final Throwable t) {
...
+            resultTable.handleBarrageError(new TableDataException("Barrage subscription error", t));
```
Comment (devinrsmith) [inline java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java:165]: "Barrage snapshot error"?

**Review by devinrsmith — DISMISSED** [review]

**Review by rcaudy — APPROVED** [review]

**Review by devinrsmith — DISMISSED** [review]

File: java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java [hunk 3]
```diff
@@ -160,8 +163,15 @@ public void onError(final Throwable t) {
...
+            final String label = tableHandle.export().table().walk(new TableLabelVisitor() {
+                @Override
+                public String visit(TicketTable ticketTable) {
+                    return BarrageSubscriptionImpl.nameForTableTicket(ticketTable);
+                }
+            });
             // this error will always be propagated to our CheckForCompletion#onError callback
-            resultTable.handleBarrageError(t);
+            resultTable.handleBarrageError(new TableDataException(
+                    String.format("Barrage snapshot error %s (%s)", logName, label), t));
```
Comment (rcaudy) [inline java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java:174]:
```suggestion
                    String.format("Barrage snapshot error for %s (%s)", logName, label), t));
```

File: java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java [hunk 4]
```diff
@@ -154,7 +162,14 @@ public void onError(final Throwable t) {
...
 
-            resultTable.handleBarrageError(t);
+            final String label = tableHandle.export().table().walk(new TableLabelVisitor() {
+                @Override
+                public String visit(TicketTable ticketTable) {
+                    return nameForTableTicket(ticketTable);
+                }
+            });
+            resultTable.handleBarrageError(new TableDataException(
+                    String.format("Barrage subscription error %s (%s)", logName, label), t));
```
Comment (rcaudy) [inline java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java:172]:
```suggestion
                    String.format("Barrage subscription for error %s (%s)", logName, label), t));
```

**Review by devinrsmith — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**